tests/test_timeout.py

```diff
@@ -396,7 +396,7 @@ async def test_task(deadline: float, loop: asyncio.AbstractEventLoop) -> None:
     loop = asyncio.get_running_loop()
     deadline = loop.time() + 1
     t = asyncio.create_task(test_task(deadline, loop))
-    loop.call_at(deadline + 0.0000000000001, t.cancel)
+    loop.call_at(deadline + 0.00000000001, t.cancel)
```

Comment:
```suggestion
    loop.call_at(deadline + 0.000001, t.cancel)
```

I think you made the value so small that you've tricked your system into passing the tests by running the cancel() before the previous task. The value needs to be big enough to ensure the task is scheduled after the timeout task, but small enough that it happens immediately after with no delay.

This test should be reliably failing. To verify the conditions, add prints to see when this cancel happens in relation to the timeout actions. The execution should be:
-> Timeout._on_timeout()
-> t.cancel()
-> Timeout._cancel()
-> Timeout._do_exit()

I suspect on your machine you've managed to change the execution order so `t.cancel()` happens first, which is what the previous test is for.